app/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,153 @@
+'use client';
+
+import { PropsWithChildren, FunctionComponent, useRef } from 'react';
+import Close from '../Close/Close';
+import Swipeable from '../OpenStreetMap/Swipeable';
+import { Transition, TransitionStatus } from 'react-transition-group';
+
+type DrawerProps = {
+    readonly children: React.ReactElement;
+    readonly className?: string;
+    readonly isOpen: boolean;
+    readonly onClose: () => void;
+    readonly closeButtonColor?: string;
+};
+
+type Position = {
+    x: string;
+    y: string;
+};
+
+const Drawer: FunctionComponent<DrawerProps> = (
+    props: PropsWithChildren<DrawerProps>
+) => {
+    const transformDuration: number = 300;
+    const backgroundDelay: number = 100;
+    const drawerContainerRef = useRef(null);
+    const drawerContainerDefaultTransitionStyle: React.CSSProperties = {
+        transform: `translateX(100%)`,
+        transition: `transform ${transformDuration}ms ease`,
+    };
+    const drawerContainerTransitionStyles: Record<
+        TransitionStatus,
+        React.CSSProperties
+    > = {
+        entering: {
+            transform: `translateX(0)`,
+        },
+        entered: {
+            transform: `translateX(0)`,
+        },
+        exiting: {
+            transform: `translateX(100%)`,
+            transition: `transform ${transformDuration}ms ease ${transformDuration + backgroundDelay}ms`,
+        },
+        exited: {
+            transform: `translateX(100%)`,
+            transition: `transform ${transformDuration}ms ease ${transformDuration + backgroundDelay}ms`,
+        },
+        unmounted: {
+            transform: `translateX(100%)`,
+            transition: `transform ${transformDuration}ms ease ${transformDuration + backgroundDelay}ms `,
+        },
+    };
+
+    const drawerRef = useRef(null);
+    const drawerDelay: number = 400;
+    const drawerDefaultTransitionStyle: React.CSSProperties = {
+        transform: `translateX(100vw)`,
+        transition: `transform ${transformDuration}ms ease `,
+    };
+    const drawerTransitionStyles: Record<
+        TransitionStatus,
+        React.CSSProperties
+    > = {
+        entering: {
+            transform: `translateX(calc(100vw - 100%))`,
+            transition: `transform ${transformDuration}ms ease ${drawerDelay}ms`,
+        },
+        entered: {
+            transform: `translateX(calc(100vw - 100%))`,
+            transition: `transform ${transformDuration}ms ease ${drawerDelay}ms`,
+        },
+        exiting: {
+            transform: `translateX(100vw)`,
+        },
+        exited: {
+            transform: `translateX(100vw)`,
+        },
+        unmounted: {
+            transform: `translateX(100vw)`,
+        },
+    };
+
+    return (
+        <Swipeable

```

Comment:
Big fan that we can now use Swipeable like this. Like we have an in-house solution for using swipe gestures which is very flipping cool. Makes me think we should be bringing in more mobile first features using Swipeable => pretty sure most of our users are coming in on mobile, in which case we have good incentive to focus on the mobile experience when it comes to improvements!

Reply:
Yeah, we can still update swipeable to do more too! Just as and when we need it to do more. For now, it seems to be pretty solid at registering swipe gestures.